### PR TITLE
Integrating Cognito API

### PR DESCRIPTION
**Pull request by abdullah5abid**

### CODE REVIEW

File: src/modules/auth/auth.controller.ts [hunk 1]
```diff
@@ -13,14 +17,62 @@ import { SigninDto } from '@modules/auth/dto/signin.dto';
...
-  ) {}
+  ) {
+
+    this.lambdaClient = new LambdaClient({
+      region: 'us-east-2',
+    });
+  }
+
+  private async invokeCreateUserLambda(data: any): Promise<any> {
+    // @TONY - Please change this url to your own API Gateway URL
```
Comment (tonytan4ever) [inline src/modules/auth/auth.controller.ts:40]: Please don't invoke lambda from API gateway.
You should invoke lambda by using the lambaClient.

File: src/modules/user/user.service.ts [hunk 2]
```diff
@@ -26,15 +26,34 @@ export class UsersService {
...
+    return await this.userRepository
+      .createQueryBuilder('users')
+      .where('users.phone = :phone')
+      .setParameter('phone', phone)
+      .getOne();
+  }
+
   async create(signupDto: SignupDto) {
-    const user = await this.getByEmail(signupDto.email);
+    const userEmail = await this.getByEmail(signupDto.email);
```
Comment (tonytan4ever) [inline src/modules/user/user.service.ts:38]: Can we use one DB query to check if email or phone exists?
i.e: if user provide both email and phone,  making only one db query to check existence.

File: src/modules/user/user.service.ts [hunk 3]
```diff
@@ -26,15 +26,34 @@ export class UsersService {
...
         'User with provided email already exists.',
       );
     }
 
+    // Check if phone number already exists
+    if (signupDto.phone) {
+      const userByPhone = await this.getByPhone(signupDto.phone);
+
+      if (userByPhone) {
+        throw new NotAcceptableException(
```
Comment (tonytan4ever) [inline src/modules/user/user.service.ts:51]: Please use `ConflictException` for Http code 409

File: src/modules/user/user.service.ts [hunk 4]
```diff
@@ -26,15 +26,34 @@ export class UsersService {
...
+      .getOne();
+  }
+
   async create(signupDto: SignupDto) {
-    const user = await this.getByEmail(signupDto.email);
+    const userEmail = await this.getByEmail(signupDto.email);
 
-    if (user) {
+    if (userEmail) {
       throw new NotAcceptableException(
```
Comment (tonytan4ever) [inline src/modules/user/user.service.ts:41]: Please use `ConflictException` for Http code 409

File: src/modules/auth/dto/signup.dto.ts [hunk 5]
```diff
@@ -1,17 +1,26 @@
 import { ApiProperty } from '@nestjs/swagger';
-import { IsEmail, IsNotEmpty, MinLength } from 'class-validator';
+import { IsEmail, IsNotEmpty, IsOptional, MinLength } from 'class-validator';
 import { SameAs } from '@modules/common/validator/same-as.validator';
 
 export class SignupDto {
   @ApiProperty({
     required: true,
   })
+  @IsOptional()
```
Comment (tonytan4ever) [inline src/modules/auth/dto/signup.dto.ts:9]: email is a mandatory field.

**Comment by abdullah5abid** [comment]
Screenshot from 2023-09-12 18-03-44
Screenshot from 2023-09-12 18-04-21
Screenshot from 2023-09-12 18-09-05
Screenshot from 2023-09-12 18-10-46
Screenshot from 2023-09-12 19-20-27
Screenshot from 2023-09-12 19-22-42
Screenshot from 2023-09-12 19-26-16
Screenshot from 2023-09-12 19-28-17

File: src/migrations/1655748241696-CreateUsersTable.ts [hunk 6]
```diff
@@ -5,7 +5,7 @@ export class CreateUsersTable1655748241696 implements MigrationInterface {
 
   public async up(queryRunner: QueryRunner): Promise<void> {
     await queryRunner.query(
-      `CREATE TABLE \`users\` (\`created_at\` datetime(6) NOT NULL DEFAULT CURRENT_TIMESTAMP(6), \`updated_at\` datetime(6) NOT NULL DEFAULT CURRENT_TIMESTAMP(6) ON UPDATE CURRENT_TIMESTAMP(6), \`id\` int NOT NULL AUTO_INCREMENT, \`firstName\` varchar(255) NOT NULL, \`lastName\` varchar(255) NOT NULL, \`email\` varchar(255) NOT NULL, \`password\` varchar(255) NOT NULL, PRIMARY KEY (\`id\`)) ENGINE=InnoDB`,
```
Comment (tonytan4ever) [inline src/migrations/1655748241696-CreateUsersTable.ts:8]: This is not correct.
A new migration should be generated to add the phone field.

**Pull request merged**